propagation_ot.go

```diff
@@ -52,9 +52,11 @@ func (p *textMapPropagator) Inject(
 	carrier.Set(zipkinTraceID, strconv.FormatUint(sc.raw.TraceID, 16))
 	carrier.Set(zipkinSpanID, strconv.FormatUint(sc.raw.SpanID, 16))
 	if !p.tracer.options.clientServerSameSpan {
-		sc.raw.ParentSpanID = 0
+		carrier.Set(zipkinParentSpanID, "0")
```

Comment:
In fixed-length binary B3 propagation, we can't represent null, so we have a convention and flag to indicate whether or not it is a root span (which ignores the parent id, treating it as null).

https://github.com/openzipkin/brave/blob/master/brave-core/src/main/java/com/github/kristofa/brave/SpanId.java#L32

In http B3 propagation, we can represent null (absence of the parent header). By serializing the value "0", which isn't null, some tracers may be fooled into thinking the span has a parent. Remember these ids are opaque bytes serialized into 64bits of hex. There's nothing special about "0"

https://github.com/openzipkin/brave/blob/master/brave-http/src/main/java/com/github/kristofa/brave/http/HttpServerRequestAdapter.java#L59